models/articles.models.js

```diff
@@ -11,20 +11,35 @@ function fetchArticleById(article_id) {
     });
 }
 
-function fetchArticles() {
-  return db
-    .query(
-      `SELECT articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, article_img_url, COUNT(*)::INT AS comment_count
-       FROM articles 
-       LEFT JOIN comments 
-       ON comments.article_id = articles.article_id
-       GROUP BY articles.article_id
-       ORDER BY articles.created_at DESC
-      `
-    )
-    .then(({ rows }) => {
-      return rows;
-    });
+function fetchArticles(topic) {
+  const queryVals = [];
+
+  let sqlQueryString = `SELECT articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, article_img_url, COUNT(*)::INT AS comment_count
+                        FROM articles 
+                        LEFT JOIN comments 
+                        ON comments.article_id = articles.article_id `;
+
+  if (topic) {
+    queryVals.push(topic);
+    sqlQueryString += `WHERE topic = $1 `;
+  }
+
+  sqlQueryString += `GROUP BY articles.article_id
+                     ORDER BY articles.created_at DESC`;
+
+  return db.query(sqlQueryString, queryVals).then(({ rows }) => {
+    return rows;
+  });
+}
+
+function checkTopicExists(topic) {
+  if (topic)
```

Comment:
Checking that the topic has been defined here is a bit unintuitive, calling a function that does nothing if the topic isn't defined is a bit difficult to follow
I think it would make sense to check if the topic is defined, and then do this function if it is
This isn't mandatory, and also more of an opinion than a definite